### PR TITLE
Spoilers

### PR DESCRIPTION
**Pull request by Zstorm999**
Partially solves issue #138 :

- spoilers are now hidden by default, they are replaced by spaces ` ` and a colored background
- a new action `s` has been added that toggles spoilers

### CODE REVIEW

File: discordmd/discordmd.go [hunk 1]
```diff
@@ -9,16 +9,45 @@ var (
...
+func ParseWithSpoilers(md string, showSpoilers bool) string {
+
 	md = boldRegex.ReplaceAllString(md, "[::b]$1[::-]")
 	md = italicRegex.ReplaceAllString(md, "[::i]$1[::-]")
 	md = underlineRegex.ReplaceAllString(md, "[::u]$1[::-]")
 	md = strikeThroughRegex.ReplaceAllString(md, "[::s]$1[::-]")
 
+	// parse spoiler tag
+	if showSpoilers {
+		md = spoilerRegex.ReplaceAllString(md, "[#E0E0E0:#383838:]$1[-:-:]")
```
Comment (sewnie) [inline discordmd/discordmd.go:34]: Aren't the colors supposed to be customized?

Comment (sewnie) [reply discordmd/discordmd.go:34]: Well, if discordo doesn't support custom colors by default.

Comment (Zstorm999) [reply discordmd/discordmd.go:34]: There are already some custom colors, for example in `ui/builder.go`, in the `buildMentions` function
```Go
if mUser.ID == clientID {
	color = "[:#5865F2]"
} else {
	color = "[#EB459E]"
}
```

**Comment by Zstorm999** [comment]
Due to recent changes in the code, I find it easier to redo this PR entirely. Will be back next with with another one.

**Pull request closed without merging**